PyInstaller/loader/rthooks/pyi_rth_multiprocessing.py

```diff
@@ -1,22 +1,23 @@
-from sys import platform as _platform
+import sys
 
 # 'spawn' multiprocessing needs some adjustments on osx
-if _platform == 'darwin':
-    import sys
+if sys.version_info >= (3, 4):
     import os
     import re
     import multiprocessing
     import multiprocessing.spawn as spawn
+    from subprocess import _args_from_interpreter_flags
 
     # prevent spawn from trying to read __main__ in from the main script
     multiprocessing.process.ORIGINAL_DIR = None
 
     def _freeze_support():
-        if 'multiprocessing.semaphore_tracker' in sys.argv[-1]:
-            m = re.compile('.*main\((\d+)\)$').match(sys.argv[-1])
-            fd = int(m.group(1))
-            import multiprocessing.semaphore_tracker
-            multiprocessing.semaphore_tracker.main(fd)
+        if len(sys.argv) >= 2 and \
+                set(sys.argv[1:-2]) == set(_args_from_interpreter_flags()) and \
+                sys.argv[-2] == '-c' and \
+                ('multiprocessing.semaphore_tracker' in sys.argv[-1] or \
+                 'multiprocessing.forkserver' in sys.argv[-1]):
```

Comment:
- This looks like magic. Please add a comment about that this code does.
- `'multiprocessing.forkserver' in sys.argv[-1]` would allow 'xxxxxmultiprocessing.forkserveryyyyyy'. Is this intended?

Reply:
I've added an extensive comment, and changed the string matching to more closely match what comes out of the Python standard library.  Unfortunately, this is quite closely linked to the private behavior of the multiprocessing module :-/  (Really, the native freeze_support should be doing this.  I'll submit a bug to upstream in a bit.)